src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs

```diff
@@ -127,6 +137,14 @@ private void StartServiceCallback(object sender, EventArgs e)
 
                     var asyncCompletionTracker = componentModel.GetService<AsyncCompletionTracker>();
                     asyncCompletionTracker.StartListening();
+
+#pragma warning disable RS0030 // Do not used banned APIs
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+#pragma warning restore RS0030 // Do not used banned APIs
```

Comment:
Not really harmful here, but RDT advising for events is explicitly free-threaded.

Reply:
➡️ This file is throwaway code (only used by legacy test project which is in the process of migration)